lls/arp.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GATEKEEPER_ARP_H_
+#define _GATEKEEPER_ARP_H_
+
+#include <rte_arp.h>
+
+#include "gatekeeper_lls.h"
+
+struct arp_cache_entry {
```

Comment:
There is no field to count how many users of this entry exit. Two blocks may ask LLS to monitor a given IP address. If one block revokes the request, the entry shouldn't be removed until the second block also revokes its request.